src/duckdb.f90

```diff
@@ -437,6 +456,42 @@ module duckdb
     type(c_ptr) :: internal_data = c_null_ptr
   end type
 
+  type, bind(c) :: duckdb_table_function
```

Comment:
Not sure these can be defined as derived types in fortran as these are not structs in c. 
These should be defined as type(c_ptr)

Reply:
Got you. I know it does not chance the result so far, but I have changed it in the last commit. I agree they are just pointers. I noticed when first started and tried the type which initially seemed to work and then I forgot about it.